src/Interpreters/ReplaceQueryParameterVisitor.cpp

```diff
@@ -68,8 +68,18 @@ const String & ReplaceQueryParameterVisitor::getParamValue(const String & name)
     auto search = query_parameters.find(name);
     if (search != query_parameters.end())
         return search->second;
-    else
-        throw Exception(ErrorCodes::UNKNOWN_QUERY_PARAMETER, "Substitution {} is not set", backQuote(name));
+    else {
+        try {
```

Comment:
`try { throw ... } catch` looks a bit silly :-) What about including all information in the exception in the first place and not using `try/catch`? Alternatively, wrap the entire block in `try` such that a single `catch` may add all context.

Reply:
Thanks for the feedback! Sorry this is my first time working on an open source project (or anything outside of class projects) I'll take a look and clean it up.